### ISSUE TITLE
Handling object construction

### ISSUE THREAD

**Issue by sharadnair** [StackExchange/Dapper#1634]
Hi team,
We use dapper to populate our generic domain models from database. This is a generic code which populates objects of a given type based on the SQL statement. We need to set some flag in the object when it is being populated from database to avoid certain operations when  populating object properties from database. Is there a "generic" way to handle object creation or get access to the object as soon as the instance is created but before any properties are populated ?
Don't want to go about making major changes to the core codebase to get this working in case there is some hooks available in Dapper. 

Many thanks
Sharad

**Comment by NickCraver** [comment]
Yep, there is! Dapper will first look for a constructor, so if you have a constructor that matches the columns, you'll get the values in and can control the hydration as you wish.
